src/vmEntry.cpp

```diff
@@ -106,6 +106,33 @@ static void* resolveMethodIdEnd() {
     return NULL;
 }
 
+// Work around for JDK-8308341, as JNI might return a none initialized VM so we check for the existence of certain VM threads 
+// "VM Thread" and "Service Thread" in this case are used as indication that the VM is in a good enough state to be used
+bool VM::checkJvmThreads() {
+    char thread_name[30];
+    int thread_name_offset = 0;
+    bool vm_thread = false, service_thread = false;
+
+    ThreadList* list = OS::listThreads();
+    while (list->hasNext()) {
+        if (!OS::threadName(list->next(), thread_name, sizeof(thread_name))) {
+            continue;
+        }
+
+        // For MacOs Thread name starts with "Java: "
+        if (strncmp(thread_name, "Java: ", 6) == 0) {
+            thread_name_offset = 6;
+        }
+
+        if (strcmp(thread_name + thread_name_offset, "VM Thread") == 0) {
```

Comment:
May cause out-of-bounds access, when thread_name_offset == 6, but thread_name does not start with "Java: "